tracer.go

```diff
@@ -140,6 +136,17 @@ func NewTracer(
 	return t, t
 }
 
+func (t *tracer) initializeHostIP() {
```

Comment:
then remove it. I am not sure why we need to keep any of these fields in the Tracer in the first place, just add the IP to the tags similar to how the hostname is added. The only possible reason is if we need the IP for Zipkin format.